주간,월간 할일 목록 보기

joi0104 opened the pull request:
#8 #21 
화면을 구성하는 Calendar view를  생성하고 그 밑에 한주간, 한달간의 할일 목록을 보는
WeekCalendar, MonthCalendar component를 생성하였습니다. 그리고 각각의 component에 DayListItem, TodoListItem을 생성하였습니다. 

또 중간에 eslint-plugin-vue를 설치해서 개발을 진행했는데 이렇게 개발 중간에 설치해서 feature 브렌치에 커밋하는게 맞는지 모르겠네요 :0 

아 그리고 여태 push했던 커밋을 기능별로 나눠서 PR을 하고 싶은데 어떻게 하는지 잘 모르겠네요.. 어쩔 수 없이 주간, 월간을 합쳐서 PR하였습니다 :(

Commit:
Create week calendar ui

Commit:
Add todoList ui in week calendar ui

Commit:
Refactoring calendar ui

Commit:
Add eslint-plugin-vue

Commit:
Refactoring app, home, calendar, login

Commit:
Refactoring week calendar ui

Commit:
Create month calendar ui

Commit:
Refactoring week calendar ui

Commit:
Refactoring month calendar ui

Commit:
Rename Login component to KakaoLoginButton component

genie-youn commented on src/views/Home.vue:
```diff
@@ -1,27 +1,31 @@
 <template>
```
이 아이는 맥락상 `home` 보다는 `login`에 가까운것 같습니다.

genie-youn commented on src/components/KakaoLoginButton.vue:
```diff
@@ -21,6 +24,7 @@ export default {
           alert('로그인에 실패했습니다.');
         }
       });
+      this.$router.push('calendar');
```
`Kakao.Auth.login` 의 `success`가 실행되고 `this.$router.push('calendar');`가 실행됨을 보장할 수 있나요?

genie-youn reviewed (commented):
나머지는 조금 있다 확인할께~

genie-youn commented on src/components/weekCalendar/WeekCalendar.vue:
```diff
@@ -0,0 +1,46 @@
+    DayListItem,
+  },
+  computed: {
+    today() {
```
computed 에서 오늘을 포함하는 이번주 날짜들이 담긴 배열 `[Date, Date, Date...]` 을 만들고 `v-for` 로 이걸 순회해서 props로 넘기는게 표현력이 더 좋을것 같습니다.

genie-youn commented on src/components/weekCalendar/WeekCalendar.vue:
```diff
@@ -0,0 +1,46 @@
+      <DayListItem
+        v-for="i in 7"
+        :key="i"
+        :day-list-item-id="getDayFromFirstDayToIdx(i-1)"
```
가급적 index를 직접 다루는 코드를 만났을 땐 더 표현력을 가질 수 있는 방법이 없을지 고민해보는것도 좋을것 같습니다.

genie-youn commented on src/components/weekCalendar/DayListItem.vue:
```diff
@@ -0,0 +1,58 @@
+<template>
```
동일한 이름의 컴포넌트는 피하는게 좋습니다.
weekly에 관련된거면 weekly 라는 prefix를 monthly엔 monthly 라는 prefix를 넣는것도 좋아보이네요.

https://kr.vuejs.org/v2/style-guide/index.html#%EA%B0%95%ED%95%9C-%EC%97%B0%EA%B4%80%EC%84%B1%EC%9D%84-%EA%B0%80%EC%A7%84-%EC%BB%B4%ED%8F%AC%EB%84%8C%ED%8A%B8-%EC%9D%B4%EB%A6%84-%EB%A7%A4%EC%9A%B0-%EC%B6%94%EC%B2%9C%ED%95%A8

genie-youn commented on src/components/weekCalendar/DayListItem.vue:
```diff
@@ -0,0 +1,58 @@
+      v-show="todoListShow"
+    >
+      <TodoListItem
+        v-for="todoListItemId in todoListItemsId"
```
todoListItemsId 보단 todoListItemsIds

genie-youn commented on src/components/weekCalendar/DayListItem.vue:
```diff
@@ -0,0 +1,58 @@
+export default {
+  name: 'DayListItem',
+  props: {
+    dayListItemId: Date,
```
프로퍼티 이름은 id 인데 타입이 Date 네요

genie-youn commented on src/components/monthCalendar/MonthCalendar.vue:
```diff
@@ -0,0 +1,71 @@
+
+export default {
+  name: 'MonthCalendar',
+  computed: {
```
동일하게 이부분 코드 읽는게 쉽지 않네요.

genie-youn reviewed (commented):
고생하셨습니다.

Commit:
Rename Home to Login

Commit:
Rename calendar,dayListItem,todoListItem using weekly and monthly prefix

Commit:
Refactoring weeklyCalendar

일주일간의 Date 배열을 만들고 배열의 item을 하나하나 weeklydayListItem에 넘겨주었습니다. 또한 idx 처리를 (item,idx) 형식을 사용해서 코드를 개선하였습니다.

Commit:
Refactoring monthlyCalendar

weeklyCalendar 와 마찬가지로 각 주의 Date 배열을 만들고 이를 넘겨주는 형식으로 코드를 개선하였습니다.

Commit:
Refactoring monthlyCalendar

아예 한달의 Date 배열을 만들어서 넘겨주는 형식으로 코드를 개선하였습니다.

Commit:
Refactoring monthlyCalendar and dayListItem

좀 더 명확한 의미를 갖는 변수명으로 바꾸고 v-for문의 인덱스를 만들어 key로 지정해주는 리팩토링을 진행하였습니다.

Commit:
Install axios plugin

axios 설치 및 기본설정을 진행하였습니다.
